processor/ratelimitprocessor/gubernator.go

```diff
@@ -407,11 +407,21 @@ func (r *gubernatorRateLimiter) getDynamicLimit(ctx context.Context,
 	}
 	// Only record the incoming hits when the current rate is within the allowed
 	// range, otherwise, do not record the hits and return the calculated rate.
-	// MaxAllowed sets a ceiling on the rate with the window duration.
+	// MaxAllowed sets a ceiling on the rate with the window duration. If the
+	// previous period had hits and the window multiplier is suggesting lowering
+	// the ingestion rate then the MaxAllowed will be allowed to go below the
+	// static rate (to as low as `1`). As soon as the window multiplier suggests
+	// increasing the ingestion rate, the MaxAllowed will jump to a minimum of
+	// static rate.
 	//
 	// NOTE(marclop) We may want to add a follow-up static ceiling to avoid
 	// unbounded growth.
-	maxAllowed := math.Max(staticRate, previous*windowMultiplier)
+	var maxAllowed float64
+	if previous > 0 && windowMultiplier <= 1 {
+		maxAllowed = max(1, previous*windowMultiplier)
```

Comment:
[**For reviewers**] I am putting a lower bound of `1` to avoid edge cases with zero ingestion rate. Let me know if it doesn't make sense OR a lower value should be used instead.